### DISCUSSION TITLE
SENSOR message not reliably sent after deep sleep.

### DISCUSSION THREAD

**Discussion by Wirewrap** [arendst/Tasmota#14991] [Q&A] [answered]
I am designing a solar powered wind meter based on ESP2866 and Tasmota 11.0.0.3
Tasmota is deep sleeping for 600 sec then wakes up, reads a speed count from a slave processor and reads wind direction from analog input. 
Power1 is pulsed bu rule1 and triggers the slave processor to send a serial message.
The serialsend starts serial bridge mode, data sent from Tasmota to slave is not used for anything.
{"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":70,"Free":441,"Rules":"ON System#Boot do Backlog SerialSend2 xyz; power1 on; power1 off ENDON"}}
A wind direction potentiometer is connected to A0.

RESULT and ANALOG are then sent like so:

MQT: tele/Wind2/RESULT = {"SerialReceived":"   X10133Y\r\n"}
MQT: tele/Wind2/SENSOR = {"Time":"2022-02-27T16:19:05","ANALOG":{"A0":508}}

This has worked well now for several days with teleperiod 10 and DeepSleepTime 600.

From yesterday  the analog tele message is not reliably sent, missing like 2 times out of 3 wakeups
Tried to set teleperiod 20, still missing as many messages.
Setting DeepSleepTime to 0 makes the analog message reliable again, once per teleperiod.

The SENSOR message is missing  in the HTML console also, so it seems it is really not sent.

What can I look at?

**Comment by Wirewrap** [comment, marked as answer]
I found my error today.
The "cmnd/Wind2/DeepSleepTime 600" was retained on the MQTT server.
Once I cleared that, operations are back to normal. 
Why is that so?

**Reply by barbudor** [reply]
Because when Tasmota sees the repeated command it restart a DeepSleep configuration.
In some case, it has the time to publish the SENSOR message, 
But sometime, processing the command is faster and gets executed before the publication.
I would expect that it is related to variations in the time taken to connect to wifi/mqtt broker.
